From d57c0f0f75b7b0e3ac4b011c97a33b5232a0aedf Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:45 -0400
Subject: [tools] perf/timechart: Add support for displaying only tasks related data

Message-id: <1408953210-26343-38-git-send-email-jolsa@redhat.com>
Patchwork-id: 88748
O-Subject: [PATCH RHEL7.1 BZ1133083 037/322] perf timechart: Add support for displaying only tasks related data
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit c87097d39dae1c42a5068e00dd3b76a4162ee0fc
Author: Stanislav Fomichev <stfomichev@yandex-team.ru>
Date: Fri Nov 1 20:25:48 2013 +0400

description
===========
In order to make SVG smaller and faster to browse add possibility to
switch off power related information with -T switch.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/Documentation/perf-timechart.txt b/tools/perf/Documentation/perf-timechart.txt
index 4373144..d844189 100644
--- a/tools/perf/Documentation/perf-timechart.txt
+++ b/tools/perf/Documentation/perf-timechart.txt
@@ -35,6 +35,9 @@ OPTIONS
 -P::
 --power-only::
         Only output the CPU power section of the diagram
+-T::
+--tasks-only::
+        Don't output processor state transitions
 -p::
 --process::
         Select the processes to display, by name or PID
diff --git a/tools/perf/builtin-timechart.c b/tools/perf/builtin-timechart.c
index 6410c9e..b3f175a 100644
--- a/tools/perf/builtin-timechart.c
+++ b/tools/perf/builtin-timechart.c
@@ -51,6 +51,7 @@ static u64  turbo_frequency;
 static u64  first_time, last_time;
 
 static bool  power_only;
+static bool  tasks_only;
 
 
 struct per_pid;
@@ -971,7 +972,8 @@ static void write_svg_file(const char *filename)
  draw_cpu_usage();
  if (proc_num)
   draw_process_bars();
- draw_c_p_states();
+ if (!tasks_only)
+  draw_c_p_states();
  if (proc_num)
   draw_wakeups();
 
@@ -1102,6 +1104,8 @@ int cmd_timechart(int argc, const char **argv,
  OPT_STRING('o', "output", &output_name, "file", "output file name"),
  OPT_INTEGER('w', "width", &svg_page_width, "page width"),
  OPT_BOOLEAN('P', "power-only", &power_only, "output power data only"),
+ OPT_BOOLEAN('T', "tasks-only", &tasks_only,
+      "output processes data only"),
  OPT_CALLBACK('p', "process", NULL, "process",
         "process selector. Pass a pid or process name.",
          parse_process),
@@ -1119,6 +1123,11 @@ int cmd_timechart(int argc, const char **argv,
  argc = parse_options(argc, argv, options, timechart_usage,
    PARSE_OPT_STOP_AT_NON_OPTION);
 
+ if (power_only && tasks_only) {
+  pr_err("-P and -T options cannot be used at the same time.\n");
+  return -1;
+ }
+
  symbol__init();
 
  if (argc && !strncmp(argv[0], "rec", 3))
-- 
1.7.1